tests/specs/session-manager.e2e.js

```diff
@@ -9,247 +12,284 @@ describe('newrelic session ID', () => {
+      await Promise.all([
+        testHandle.expectRum(),
+        browser.url(await testHandle.assetURL('session-entity.html', config)) // Setup expects before loading the page
+      ])
 
-      const ls1 = await browser.execute(getLocalStorageData)
+      const { localStorage: ls1 } = await browser.getAgentSessionInfo()
       expect(ls1).toEqual(anySession())
 
-      await browser.refresh()
-      await browser.waitForFeature('loaded')
+      await Promise.all([
+        testHandle.expectRum(),
+        browser.refresh() // Setup expects before loading the page
+      ])
 
-      const ls2 = await browser.execute(getLocalStorageData)
+      const { localStorage: ls2 } = await browser.getAgentSessionInfo()
       expect(ls2).toEqual(anySession())
       expect(ls2.value).toEqual(ls1.value)
       expect(ls2.expiresAt).toEqual(ls1.expiresAt)
-      await browser.execute(function () { window.localStorage.clear() })
     })
 
     it('should keep a session id across page loads - Same tab navigation', async () => {
-      await browser.url(await testHandle.assetURL('session-entity.html', { init }))
-      await browser.waitForFeature('loaded')
+      await Promise.all([
+        testHandle.expectRum(),
+        browser.url(await testHandle.assetURL('session-entity.html', config)) // Setup expects before loading the page
+      ])
 
-      const ls1 = await browser.execute(getLocalStorageData)
+      const { localStorage: ls1 } = await browser.getAgentSessionInfo()
       expect(ls1).toEqual(anySession())
 
-      await browser.url(await testHandle.assetURL('fetch.html', { init }))
-      await browser.waitForFeature('loaded')
+      await Promise.all([
+        testHandle.expectRum(),
+        browser.url(await testHandle.assetURL('fetch.html', config)) // Setup expects before loading the page
+      ])
 
-      const ls2 = await browser.execute(getLocalStorageData)
+      const { localStorage: ls2 } = await browser.getAgentSessionInfo()
       expect(ls2).toEqual(anySession())
       expect(ls2.value).toEqual(ls1.value)
       expect(ls2.expiresAt).toEqual(ls1.expiresAt)
-      await browser.execute(function () { window.localStorage.clear() })
     })
 
-    it('should keep a session id across page loads - Multi tab navigation', async () => {
-      await browser.url(await testHandle.assetURL('session-entity.html', { init }))
-      await browser.waitForFeature('loaded')
+    withBrowsersMatching(supportsMultipleTabs)('should keep a session id across page loads - Multi tab navigation', async () => {
+      await Promise.all([
+        testHandle.expectRum(),
+        browser.url(await testHandle.assetURL('session-entity.html', config)) // Setup expects before loading the page
+      ])
 
-      const ls1 = await browser.execute(getLocalStorageData)
+      const { localStorage: ls1 } = await browser.getAgentSessionInfo()
       expect(ls1).toEqual(anySession())
 
-      await browser.execute(function () { window.open('/') })
```

Comment:
I think `window.open` already worked on all envs.  Are we limiting ourselves by doing this around a browser matcher arbitrarily?

Reply:
In my testing, `window.open` does not work in mobile browsers.

Reply:
So I did some more testing and here are my findings. While the original test appears to pass on ios, a new window/tab is not actually being opened. I have also tried creating a new link element and clicking it using `browser.execute` and `$().click()`. Nothing I can find can actually force ios safari to open a new tab or window.